JsonLogic.Tests/GithubTests.cs

```diff
@@ -12,6 +13,10 @@ namespace Json.Logic.Tests;
 
 public class GithubTests
 {
+#if NET8_0_OR_GREATER
+	private readonly JsonTypeInfo<Rule> _ruleTypeInfo = (JsonTypeInfo<Rule>)Rule.JsonTypeResolver.GetTypeInfo(typeof(Rule), JsonSerializerOptions.Default)!;
```

Comment:
Instead of a JsonTypeInfo field, we could make a JsonSerializerOptions and then the below calls can be options.Read<Rule>(jsonRule) which would also remove the `#if` conditions.